python/src/rest_api/aac_rest_app.py

```diff
@@ -0,0 +1,569 @@
+
+
+# AaC Plugin Commands
+
+
+@app.get("/commands", status_code=HTTPStatus.OK, response_model=list[CommandModel])
+def get_aac_commands() -> list[CommandModel]:
+    """
+    Return a list of all available plugin commands.
+
+    Global Args:
+        ACTIVE_CONTEXT (LanguageContext): The global active language context.
+
+    Returns:
+        A list of CommandModel objects
+    """
+    aac_and_plugin_commands = _get_rest_api_compatible_commands()
+    return [to_command_model(command) for command in aac_and_plugin_commands.values()]
+
+
+@app.post("/command", status_code=HTTPStatus.OK, response_model=CommandResponseModel)
+def execute_aac_command(command_request: CommandRequestModel):
+    """
+    Execute the command and return the result.
+
+    Global Args:
+        ACTIVE_CONTEXT (LanguageContext): The global active language context.
+
+    Args:
+        command_request (CommandRequestModel): The AaC command to be executed.
+
+    Returns:
+        A CommandResponseModel object containing the result of the command execution.
+    """
+    aac_commands_by_name = _get_rest_api_compatible_commands()
+    aac_command = aac_commands_by_name.get(command_request.name)
+
+    if aac_command is not None:
+        aac_command_argument_names = [arg.name for arg in aac_command.arguments]
+        arguments = [arg for arg in command_request.arguments if arg not in aac_command_argument_names]
+
+        try:
+            result = aac_command.callback(*(arguments or []))
+            success = result.status_code == ExecutionStatus.SUCCESS
+            result_message = f"{result.name}: {result.status_code.name.lower()}\n\n{result.get_messages_as_string()}"
+        except Exception as error:
+            success = False
+            result_message = f"{result.name}: failure\n\n{error}"
+        finally:
+            return CommandResponseModel(command_name=aac_command.name, result_message=result_message, success=success)
+    else:
+        _report_error_response(
+            HTTPStatus.NOT_FOUND,
+            f"Command name {command_request.name} not found in the list of available commands: {list(aac_commands_by_name.keys())}.",
+        )
+
+
+def _get_available_files_in_workspace() -> set[AaCFile]:
+    """
+    Get the available AaC files in the workspace sans files already in the context.
+
+    Global Args:
+        ACTIVE_CONTEXT (LanguageContext): The global active language context.
+
+    Returns:
+        A set containing available AaC files in workspace.
+    """
+    aac_files_in_context = set(_get_files_in_context())
+    aac_files_in_workspace = set(find_aac_files(WORKSPACE_DIR))
+
+    return aac_files_in_workspace.difference(aac_files_in_context)
+
+
+async def refresh_available_files_in_workspace(context: LanguageContext) -> None:
+    """
+    Used to refresh the available files. Used in async since it takes too long for being used in request-response flow.
+
+    Global Args:
```

Comment:
```suggestion
    Args:
```
This one won't be the global context yet, this is the local to method being assigned to the global instance.